pkg/controllers/allocation/controller.go

```diff
@@ -66,13 +86,15 @@ func NewController(kubeClient client.Client, coreV1Client corev1.CoreV1Interface
 // Reconcile executes an allocation control loop for the resource
 func (c *Controller) Reconcile(ctx context.Context, object client.Object) (reconcile.Result, error) {
 	provisioner := object.(*v1alpha2.Provisioner)
+	c.batcher.Wait(fmt.Sprintf("%s/%s", provisioner.Name, provisioner.Namespace))
```

Comment:
Do we need to wait inside the reconcile loop? Is it possible that this will trigger double reconciles? The first event will trigger this and then start waiting. All subsequent events will queue up. Once this wait finishes, the loop will execute and then ctrl will trigger this even due to the queued events.

Reply:
It shouldn't based on the docs (https://pkg.go.dev/k8s.io/client-go@v0.21.2/util/workqueue, since it's "Stingy") and from real world tests, it does not reprocess. I think I need to dig into this more to see exactly why this is the case though. I'm not entirely sure how it would know if a pod is queued up after we list pods in a reconcile.

Reply:
There are some nice benefits of putting this in the reconcile loop itself, namely simplicity/linearity of logic. In general, I much prefer to put filtering logic in here, rather than in the Watch()

Reply:
Yeah, agreed, putting it in the Watch is basically a non-starter since it's completely synchronous.  We wouldn't be able to do provisioner based batching as efficiently.

Reply:
I think we might be able to scrounge up the event queue somehow and pass it in. Then we could enqueue. Probably cleaner this way though.